services/delegation/src/modules/blockchain/blockchain.service.ts

```diff
@@ -110,6 +124,43 @@ export class BlockchainService {
     return false;
   }
 
+  async undelegate(address: string, amount: ethers.BigNumber) {
+    try {
+      const tx = await this.contract.undelegate(address, {
+        gasPrice: this.provider.getGasPrice(),
+        value: amount,
```

Comment:
`amount` is a param to the function, we don't want to send those tokens to the contract.

Reply:
Yes, you're right, I missed this one.